usecases/configuration.go

```
@@ -109,66 +110,60 @@ func Configure(sys *components.System) ([]json.RawMessage, error) {
+		err = enc.Encode(defaultConfig) // Write default values into systemconfig since file was empty
 		if err != nil {
-			return rawBytes, err
+			return nil, fmt.Errorf("error writing default values to system config")
 		}
-		return rawBytes, fmt.Errorf("a new configuration file has been written with %d bytes. Please update it and restart the system", nBytes)
+		return nil, ErrNewConfig
 	}
 
-	// the system configuration file could be open, read the configurations and pass them on to the system
-	defer systemConfigFile.Close()
-	configBytes, err := os.ReadFile("systemconfig.json")
+	var configurationIn configFileIn
+	err = json.NewDecoder(systemConfigFile).Decode(&configurationIn) // Read the contents of systemconfig into configurationIn
 	if err != nil {
-		return rawBytes, err
+		return nil, fmt.Errorf("error reading systemconfig: %v", err)
 	}
 
-	// the challenge is that the definition of the unit asset is unknown to the mbaigo library and only known to the system that invokes the library
-	var configurationIn configFileIn
-	// extract the information related to the system separately from the unit_assets (i.e., the resources)
-	type Alias configFileIn
-	aux := &struct {
-		Resources []json.RawMessage `json:"unit_assets"`
-		*Alias
-	}{
-		Alias: (*Alias)(&configurationIn),
-	}
-	if err := json.Unmarshal(configBytes, aux); err != nil {
-		return rawBytes, err
-	}
-	if len(aux.Resources) > 0 {
-		configurationIn.rawResources = aux.Resources
+	var rawResources []json.RawMessage
+	if len(configurationIn.Resources) > 0 { // If unit assets was present in systemconfig file, send those
+		rawResources = configurationIn.Resources
 	} else {
-		var rawMessages []json.RawMessage
-		for _, s := range defaultConfig.Assets {
-			// convert the struct to JSON-encoded byte array
+		for _, s := range defaultConfig.Assets { // Otherwise send the system default
 			jsonBytes, err := json.Marshal(s)
 			if err != nil {
 				fmt.Println("Failed to marshal struct:", err)
```

Return error instead of printing

---

Good catch, missed that print